module/apmawssdkgo/session_test.go

```diff
@@ -91,3 +92,59 @@ func TestS3(t *testing.T) {
 		assert.Equal(t, tx.ID, span.ParentID)
 	}
 }
+
+func TestDynamoDB(t *testing.T) {
+	region := "us-west-2"
+	cfg := aws.NewConfig().
+		WithRegion(region).
+		WithDisableSSL(true).
+		WithCredentials(credentials.AnonymousCredentials)
+
+	spanSubtype := "dynamodb"
+	spanType := serviceTypeMap[spanSubtype]
```

Comment:
According to the [database spec](https://github.com/elastic/apm/blob/master/specs/agents/tracing-instrumentation-db.md#database-span-typesubtype):
 - `type` should be "db"
 - `subtype` should be the database name (e.g. "dynamodb")

IIANM, `spanType` here will be "dynamodb". In general I think we should avoid referring to internals of the implementation like this, as I find it makes issues like this less obvious. Instead, use literals for the expected values.

Reply:
ah good catch, I got it mixed up with `db.type`.

>In general I think we should avoid referring to internals of the implementation like this, as I find it makes issues like this less obvious. Instead, use literals for the expected values.

I'm not sure I'm understanding. You mean to not use an incoming value like `req.ClientInfo.ServiceName` as the subtype, but match on it/some indicator in the request, and then return specific literals of what we want?

Reply:
no sorry, I just meant in the tests to stop referring to `serviceTypeMap`, and instead do:

```go
assert.Equal(t, "db", span.Type)
assert.Equal(t, "dynamodb", span.Subtype)
```

where possible I prefer the tests to be external test packages (apmawssdkgo_test) to force the separation

Reply:
ahhh ok, got it